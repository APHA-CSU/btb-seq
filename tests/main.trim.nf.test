nextflow_process {

    name "Test Process trim"
    script "main.nf"
    process "trim"

    test("Should run trimmomatic on input fastq files") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = ["phix-m", file("${projectDir}/tests/data/phix_test/dedup_1.fastq"), file("${projectDir}/tests/data/phix_test/dedup_2.fastq"), file("${projectDir}/references/adapter.fasta")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
