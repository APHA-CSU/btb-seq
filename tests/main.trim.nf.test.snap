{
    "Should run trimmomatic on input fastq files": {
        "content": [
            {
                "0": [
                    [
                        "phix-m",
                        "trimmed_1.fastq:md5,0772e0ad75f321e8b478fd7d4a8e4248",
                        "trimmed_2.fastq:md5,d069fdf6066ba6d941ed2ffee4b79640"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-02T23:31:58.785781745"
    }
}